client/my-sites/email/inbox/index.jsx

```diff
@@ -0,0 +1,93 @@
+import emailIllustration from 'calypso/assets/images/email-providers/email-illustration.svg';
+import EmptyContent from 'calypso/components/empty-content';
+import Main from 'calypso/components/main';
+import PromoCard from 'calypso/components/promo-section/promo-card';
+import { hasEmailForwards } from 'calypso/lib/domains/email-forwarding';
+import { hasPaidEmailWithUs } from 'calypso/lib/emails';
+import { getGSuiteMailboxCount } from 'calypso/lib/gsuite';
+import { getConfiguredTitanMailboxCount } from 'calypso/lib/titan';
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import EmailManagementHome from 'calypso/my-sites/email/email-management/email-home';
+import MailboxSelectionList from 'calypso/my-sites/email/inbox/mailbox-selection-list';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import { getDomainsBySiteId } from 'calypso/state/sites/domains/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+import './style.scss';
+
+const NoAccessCard = () => {
+	const translate = useTranslate();
+
+	return (
+		<Main>
+			<EmptyContent
+				title={ translate( 'You are not authorized to view this page' ) }
+				illustration={ '/calypso/images/illustrations/illustration-404.svg' }
+			/>
+		</Main>
+	);
+};
+
+const MainHeader = () => {
+	const translate = useTranslate();
+	const image = {
+		path: emailIllustration,
+		align: 'right',
+	};
+
+	return (
+		<>
+			<PromoCard
+				className={ 'inbox__is-inbox-card' }
+				icon={ null }
+				image={ image }
+				isPrimary={ true }
+				title={ translate( 'Pick a domain to get started' ) }
+			>
+				<p>{ translate( 'Pick one of your domains below to add an email solution.' ) }</p>
+			</PromoCard>
+		</>
+	);
+};
+
+const hasAtLeastOneMailbox = ( domains ) =>
+	domains.some(
+		( domain ) =>
+			getGSuiteMailboxCount( domain ) > 0 || getConfiguredTitanMailboxCount( domain ) > 0
+	);
+
+const showActiveDomainList = ( domains ) =>
+	domains.some( ( domain ) => hasPaidEmailWithUs( domain ) || hasEmailForwards( domain ) );
+
+const InboxManagement = () => {
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const canManageSite = useSelector( ( state ) =>
+		canCurrentUser( state, selectedSiteId, 'manage_options' )
+	);
+	const domains = useSelector( ( state ) => getDomainsBySiteId( state, selectedSiteId ) );
+	const translate = useTranslate();
+
+	if ( ! canManageSite ) {
+		return <NoAccessCard />;
+	}
+
+	const nonWPCOMDomains = domains.filter( ( domain ) => ! domain.isWPCOMDomain );
+
+	if ( hasAtLeastOneMailbox( nonWPCOMDomains ) ) {
+		return <MailboxSelectionList />;
+	}
+
+	return (
+		<CalypsoShoppingCartProvider>
+			<EmailManagementHome
+				emailListInactiveHeader={ <MainHeader /> }
+				sectionHeaderLabel={ translate( 'Domains' ) }
+				showActiveDomainList={ showActiveDomainList( nonWPCOMDomains ) }
+			/>
+		</CalypsoShoppingCartProvider>
+	);
```

Comment:
Unrelated to those changes but I don't think this page should be responsible for rendering the `Emails` page (which in turn will render the `Email Comparison` page). I think we should be redirecting the user to that page instead, something that should make the code of this `InboxManagement` simpler. Note right now this also leads to minor discrepancies in the design (white vs gray background, header alignment):

`/inbox` | `/email`
------ | -----
<img width="624" alt="before" src="https://user-images.githubusercontent.com/594356/136007693-6c1877a9-0420-497f-8d29-7f294f7e7041.png"> | <img width="623" alt="after" src="https://user-images.githubusercontent.com/594356/136007703-36fd19d7-04db-4087-8494-bf45bb9b1ecf.png">

Reply:
I agree with you totally. There might be need to resolve some history issues from redirection, but let us refine in the next iteration!